src/components/structures/login/Login.js

```diff
@@ -257,6 +276,125 @@ module.exports = React.createClass({
         });
     },
 
+    _tryWellKnownDiscovery: async function(serverName) {
+        if (!serverName.trim()) {
+            // Nothing to discover
+            this.setState({discoveryError: "", discoveredHsUrl: "", discoveredIsUrl: ""});
+            return;
+        }
+
+        try {
+            const wellknown = await this._getWellKnownObject(`https://${serverName}/.well-known/matrix/client`);
+            if (!wellknown["m.homeserver"]) {
+                console.error("No m.homeserver key in well-known response");
+                this.setState({discoveryError: _t("Invalid homeserver discovery response")});
+                return;
+            }
+
+            const hsUrl = this._sanitizeWellKnownUrl(wellknown["m.homeserver"]["base_url"]);
+            if (!hsUrl) {
+                console.error("Invalid base_url for m.homeserver");
+                this.setState({discoveryError: _t("Invalid homeserver discovery response")});
+                return;
+            }
+
+            console.log("Verifying homeserver URL: " + hsUrl);
```

Comment:
Presumably this would go if https://github.com/matrix-org/matrix-doc/pull/1700 is approved?

Reply:
Yup